src/fsharp/PatternMatchCompilation.fs

```diff
@@ -793,11 +796,37 @@ let CompilePatternBasic
                     mkReraise matchm resultTy
 
                 | Throw ->
-                    // We throw instead of rethrow on unmatched try-catch in a computation expression. But why?
-                    // Because this isn't a real .NET exception filter/handler but just a function we're passing
+                    // We use throw, or EDI.Throw when EDI is supported, instead of rethrow on unmatched try-catch in a computation expression.
+                    // But why? Because this isn't a real .NET exception filter/handler but just a function we're passing
                     // to a computation expression builder to simulate one.
-                    mkThrow matchm resultTy (exprForVal matchm origInputVal)
-
+                    let ediThrowMethInfo =
+                        // EDI.Throw: exn -> unit
+                        let name, sigTys = "Throw", [ g.exn_ty; g.unit_ty ]
+
+                        g.system_ExceptionDispatchInfo_ty
+                        |> Option.bind (fun ty ->
+                            TryFindIntrinsicMethInfo infoReader matchm (AccessorDomain.AccessibleFromEverywhere) name ty
+                            |> List.tryFind (fun methInfo ->
+                                not methInfo.IsInstance &&
+                                (
+                                    match methInfo.GetParamTypes(amap, matchm, []) with
+                                    | [] -> false
+                                    | argTysList ->
+                                        let argTys = (argTysList |> List.reduce (@)) @ [ methInfo.GetFSharpReturnTy (amap, matchm, []) ] in
+                                        if argTys.Length <> sigTys.Length then
+                                            false
+                                        else
+                                            (argTys, sigTys)
+                                            ||> List.forall2 (typeEquiv g)
+                                )
+                            ))
+
+                    match ediThrowMethInfo with
+                    | None ->
+                        mkThrow matchm resultTy (exprForVal matchm origInputVal)
+                    | Some ediThrowMethInfo ->
+                        fst <| BuildMethodCall tcVal g amap NeverMutates matchm false
```

Comment:
no `<|` in compiler please, except specific idioms  :-))

Reply:
Yes might need to place this in a `mkSequential` with the previous thing generated, just to get the return type right